rust/agents/validator/src/submit.rs

```diff
@@ -51,11 +57,54 @@ impl ValidatorSubmitter {
             mailbox_domain: self.mailbox.domain().id(),
             storage_location: self.checkpoint_syncer.announcement_location(),
         };
-        let signed_announcement = self.signer.sign(announcement).await?;
+        let signed_announcement = self.signer.sign(announcement.clone()).await?;
         self.checkpoint_syncer
             .write_announcement(&signed_announcement)
             .await?;
 
+        // Ensure that the validator has announced themselves before we enter
+        // the main validator submit loop. This is to avoid a situation in
+        // which the validator is signing checkpoints but has not announced
+        // their locations, which makes them functionally unusable.
+        let validators: [H256; 1] = [self.signer.eth_address().into()];
+        loop {
+            info!("Checking for validator announcement");
+            if let Some(locations) = self
+                .validator_announce
+                .get_announced_storage_locations(&validators)
+                .await?
+                .first()
+            {
+                if locations.contains(&self.checkpoint_syncer.announcement_location()) {
+                    info!("Validator has announced signature storage location");
+                    break;
+                }
+                info!("Validator has not announced signature storage location");
+                let balance_delta = self
+                    .validator_announce
+                    .announce_tokens_needed(signed_announcement.clone())
+                    .await?;
+                if balance_delta.cmp(&U256::zero()) == std::cmp::Ordering::Greater {
+                    warn!(
+                        "Please send {} tokens to the validator address {} to announce",
+                        balance_delta, announcement.validator,
+                    );
```

Comment:
Filling in values to strings makes it much harder in GCP logs to filter by a log type

Also you need address to use `Debug` not `Display` or it will truncate the address.
```suggestion
                    warn!(
                        tokens_needed=%balance_delta,
                        validator_address=?announcement.validator,
                        "Please send tokens to the validator address to announce",
                    );
```